File: GXpert/GXpert.Web/Modules/ServerTypes/Playlist/PlayListForm.ts
import { StringEditor, LookupEditor, TextAreaEditor, IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface PlayListForm {
    Title: StringEditor;
    ClassId: LookupEditor;
    SubjectId: LookupEditor;
    TeacherId: LookupEditor;
    Description: TextAreaEditor;
    PlayListFilePath: StringEditor;
    PlayListVersion: IntegerEditor;
}

export class PlayListForm extends PrefixedContext {
    static readonly formKey = 'Playlist.PlayList';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PlayListForm.init)  {
            PlayListForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;
            var w2 = TextAreaEditor;
            var w3 = IntegerEditor;

            initFormType(PlayListForm, [
                'Title', w0,
                'ClassId', w1,
                'SubjectId', w1,
                'TeacherId', w1,
                'Description', w2,
                'PlayListFilePath', w0,
                'PlayListVersion', w3
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Playlist/PlayList/PlayListForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Playlist.Forms;

[FormScript("Playlist.PlayList")]
[BasedOnRow(typeof(PlayListRow), CheckNames = true)]
public class PlayListForm
{
    public string Title { get; set; }
    
    [HalfWidth]
    public int ClassId { get; set; }
    [HalfWidth]
    public int SubjectId { get; set; }
    [HalfWidth]
    public int TeacherId { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }
    public string PlayListFilePath { get; set; }
    public int PlayListVersion { get; set; }
}